test/SourceKit/CursorInfo/doc_attr.swift

```diff
@@ -0,0 +1,9 @@
+@_documentation(visibility: public)
+enum E
+{
+}
+
+// The @_documentation attribute caused sourcekit-lsp to crash
+// cf. https://github.com/apple/swift/issues/64309
+
+// RUN: %sourcekitd-test -req=cursor -pos=2:6 %s -- %s
```

Comment:
Where in the response is `@_documentation` showing up? If it's user inaccessible, I wouldn't have expected it in cursor info either.

Reply:
@bnbarham I added some output checking in this test; it seems to be showing up in the XML output. I'm not sure if this is expected, but i've updated the test.

Reply:
That is definitely very unexpected to me. IIUC the annotated description should not be including user inaccessible attributes. I guess we're missing adding the `PrintUserInaccessibleAttrs` in `printQuickHelpDeclaration`. Ideally we'd fix that as well, but if you don't want to do that it's fine.

If we really want a test for this I'd probably prefer something like a -dump-ast (assuming it includes user inaccessible?) in test/attr. Once we fix cursor info this test would be invalid otherwise.

Reply:
Adding the test in test/attr makes sense as a way to ensure that the attribute gets printed correctly, but i'd like to leave the test here (without the output checking) since that was the original error case, if that's alright.

Reply:
If you're changing the test anyway, I'd strongly prefer just not including it. The crash has a known cause that we can test, so I don't think there's much use in testing the crash itself.

Reply:
It looks like `-dump-ast` doesn't include UserInaccessible attributes either:

```
(source_file "/Users/victoria_m/swift-base/swift/test/attr/attr_documentation.swift"
  (enum_decl range=[/Users/victoria_m/swift-base/swift/test/attr/attr_documentation.swift:5:1 - line:7:1] "E" interface type='E.Type' access=internal non-resilient))
```

Reply:
Ah that's a pity 😓. You can keep the cursor info test without output if you want. Really we should check that inaccessible attributes don't get printed in generated interface/completion/cursor info, but I don't expect you to do that.

Reply:
I took the output check out of the test. The test as a whole could be taken out or modified when the UserInaccessible filtering is added in the future.